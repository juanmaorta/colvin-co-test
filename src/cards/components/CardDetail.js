import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import { withStyles } from '@material-ui/core/styles'

import styles from '../../styles'

import BlackCard from './BlackCard'
import WhiteCard from './WhiteCard'

class CardDetail extends Component {
  render () {
    const {
      card,
      classes,
      addCard,
      deleteCard,
      editCard
    } = this.props

    return (
      <div className={classNames(classes.layout, classes.cardGrid)}>
        <BlackCard
          card={card}
          editable={false}
        />
        <Grid container spacing={40}>
          {card.whiteCards.map((wc, idx) => {
            return (
              <WhiteCard
                card={wc}
                handleDelete={deleteCard}
                handleEdit={editCard}
                key={idx}
              />
            )
          })}
        </Grid>
        <div className={classes.buttons}>
          <Button variant='contained' onClick={addCard}>Add a white card</Button>
        </div>
      </div>
    )
  }
}

CardDetail.propTypes = {
  card: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(CardDetail)
