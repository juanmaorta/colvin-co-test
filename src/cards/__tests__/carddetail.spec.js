/* global expect, describe, it, xit */
import React from 'react'
import { shallow } from 'enzyme'

import CardDetail from '../components/CardDetail'
import * as stubs from './stubs'

describe('CardDetail snapshots', () => {
  it('should match for CardDetail component', () => {
    const wrapper = shallow(
      <CardDetail
        classes={stubs.classes}
        card={stubs.card}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
