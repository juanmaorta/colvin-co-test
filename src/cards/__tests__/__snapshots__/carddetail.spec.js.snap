// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetail snapshots should match for CardDetail component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(CardDetail)
    card={
        Object {
            "id": 1,
            "text": "Goldfinger: He's the man, the man with...",
          }
    }
    classes={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "card": Object {
        "id": 1,
        "text": "Goldfinger: He's the man, the man with...",
      },
      "classes": Object {
        "appBar": "CardDetail-appBar-1",
        "blackBackGround": "CardDetail-blackBackGround-5",
        "blackCardContent": "CardDetail-blackCardContent-6",
        "blackCardTitle": "CardDetail-blackCardTitle-3",
        "buttons": "CardDetail-buttons-2",
        "card": "CardDetail-card-4",
        "cardGrid": "CardDetail-cardGrid-12",
        "footer": "CardDetail-footer-7",
        "heroButtons": "CardDetail-heroButtons-10",
        "heroContent": "CardDetail-heroContent-9",
        "heroUnit": "CardDetail-heroUnit-8",
        "layout": "CardDetail-layout-11",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "card": Object {
          "id": 1,
          "text": "Goldfinger: He's the man, the man with...",
        },
        "classes": Object {
          "appBar": "CardDetail-appBar-1",
          "blackBackGround": "CardDetail-blackBackGround-5",
          "blackCardContent": "CardDetail-blackCardContent-6",
          "blackCardTitle": "CardDetail-blackCardTitle-3",
          "buttons": "CardDetail-buttons-2",
          "card": "CardDetail-card-4",
          "cardGrid": "CardDetail-cardGrid-12",
          "footer": "CardDetail-footer-7",
          "heroButtons": "CardDetail-heroButtons-10",
          "heroContent": "CardDetail-heroContent-9",
          "heroUnit": "CardDetail-heroUnit-8",
          "layout": "CardDetail-layout-11",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
